backoffice/views.py

```diff
@@ -58,7 +76,44 @@ def courses_list(request):
     if pattern:
         courses = [course for course in courses
                 if pattern in course.title
-                or pattern in course.id.to_deprecated_string()]
+                or pattern in course.course.id.to_deprecated_string()]
+
+    if request.method == 'POST':
```

Comment:
Je comprend. Alors je propose d'écrire une fonction : 

    def get_filtered_course_infos(request):
        pattern = request.GET.get('search')
        # Il va falloir renommer la fonction course_infos en get_course_info.
        course_infos = [course_infos(course) for course in get_courses(request.user)]
        if pattern:
            ...
        return course_infos